Recreate existing CFN template using CDK constructs for stripe-webhook-endpoints

LAKSHMIRPILLAI opened the pull request:
## What does this change?

This PR is to recreate existing CFN template using CDK constructs, while retaining existing infrastructure for stripe-webhook-endpoints.

For apps where avoiding downtime is critical, a dual-stack migration strategy forGuCDK Migration is recommended . The idea is to duplicate the key parts of the stack using the pattern, test, and, once confident, switch DNS to point to the new stack.
In this phase we create a new version of your app's infrastructure, according to CDK’s best practices. So, we should have a functional copy of the application, which can be used for testing purposes. Riff-Raff will have been configured to deploy to both versions of the infrastructure, so you can operate in this state for as long as you need to.

[Trello card](https://trello.com/c/mXJhc0b9/255-migration-of-stripe-webhook-endpoints-cloudformation-recreate-existing-cfn-template-using-cdk-constructs-while-retaining-existin)


In more detail, this change:

Makes the lambdas using cdk constructs
Adds an API gateway endpoint for calling it
Adds necessary config for it to be called from stripe accounts (GNM Membership)
Changes existing stripe service calls to specify which account they’re calling (so that, for example, we can look up a customer in the right account, depending on which account’s webhook called the lambda)
We’ve tested this by deploying it to CODE, creating the necessary parameter store config for CODE, and triggering events on the stripe accounts in test mode, which show the calls succeeding.


## Post-Deployment Steps

Once this change is deployed, there are some post-deployment steps, which need to happen in order:

- Create a (non-test-mode) webhook (initially disabled) to call the payment-intent-issue event lambda:
  - [ ] On the GNM Membership account
- Create a (non-test-mode) webhook (initially disabled) to call the customer update event lambda:
  - [ ] On the GNM Membership account
- Add the following AWS Parameter Store entries
  - [x] /CODE/support/stripe-webhook-endpoints/endpointSecret
  - [x] /CODE/support/stripe-webhook-endpoints/zuoraBaseUrl
  - [x] /CODE/support/stripe-webhook-endpoints/zuoraClientId
  - [x] /CODE/support/stripe-webhook-endpoints/zuoraSecret
  - [x] /PROD/support/stripe-webhook-endpoints/endpointSecret 
  - [x] /PROD/support/stripe-webhook-endpoints/zuoraBaseUrl
  - [x] /PROD/support/stripe-webhook-endpoints/zuoraClientId 
  - [x] /PROD/support/stripe-webhook-endpoints/zuoraSecret 
  - [ ] Enable the new webhooks
- Remove these AWS Parameter Store entries:
  - [ ] /CODE/membership/payment-intent-issues/endpointSecret
  - [ ] /CODE/membership/payment-intent-issues/zuoraBaseUrl
  - [ ] /CODE/membership/payment-intent-issues/zuoraClientId
  - [ ] /CODE/membership/payment-intent-issues/zuoraSecret
  - [ ] /PROD/membership/payment-intent-issues/endpointSecret
  - [ ] /PROD/membership/payment-intent-issues/zuoraBaseUrl
  - [ ] /PROD/membership/payment-intent-issues/zuoraClientId
  - [ ] /PROD/membership/payment-intent-issues/zuoraSecret

michaelbjacobson commented on cdk/lib/stripe-webhook-endpoints.ts:
```diff
@@ -1,15 +1,225 @@
+
+        // ---- Existing CFN template ---- //
+        const yamlTemplateFilePath = path.join(__dirname, "../..", "handlers/stripe-webhook-endpoints/cfn.yaml");
+        const yamlDefinedResources =  new CfnInclude(this, "YamlTemplate", {
```
It's really not a problem, but there's no need to assign the `CfnInclude` to a `const` or `var` here, unless you need to reference it later on.

michaelbjacobson reviewed (approved):
Great work on this! When I glanced over it this morning I thought you needed to make an additional change in the CI workflow, but now I see that it's all set up perfectly.

✅ ✅ ✅

tomrf1 commented:
@LAKSHMIRPILLAI this is causing the cdk build to fail on main because of an unused value -
https://github.com/guardian/support-service-lambdas/actions/runs/8373319007/job/22926129132#step:6:61

LAKSHMIRPILLAI commented:
> @LAKSHMIRPILLAI this is causing the cdk build to fail on main because of an unused value -
> https://github.com/guardian/support-service-lambdas/actions/runs/8373319007/job/22926129132#step:6:61

Yes , I updated that in the other PR and then when I ran the synth step, the tests failed, and when I updated the snapshots   the java version jumped to 21.
